pkg/v1/image.go

```diff
@@ -25,6 +25,11 @@ type Image interface {
 	// The order of the list is oldest/base layer first, and most-recent/top layer last.
 	Layers() ([]Layer, error)
 
+	// CompressedLayers returns the ordered collection of compressed filesystem layers that
+	// comprise this image. The order of the list is oldest/base layer first, and
+	// most-recent/top layer last. CompressedLayers are useful for optimized pushing.
+	CompressedLayers() ([]Layer, error)
```

Comment:
Why is this better than just calling `Compressed` on each layer that gets returned? It's unclear to me why this is needed.

Reply:
Through profiling today, I found out that `Layers()` was called multiple times and returned _new_ layers. So the speedup was because `CompressedLayers` cached the compressed layers, while `Layers` did not.